backend/tests/test_app.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+from fastapi.testclient import TestClient
+from freezegun import freeze_time
+from sqlmodel import Session, SQLModel, create_engine
+from sqlmodel.pool import StaticPool
+
+from app.main import app, get_session
+
+
+@pytest.fixture(name="session")
+def session_fixture():
+    engine = create_engine(
+        "sqlite://", connect_args={"check_same_thread": False}, poolclass=StaticPool
+    )
+    SQLModel.metadata.create_all(engine)
+    with Session(engine) as session:
+        yield session
+
+
+@pytest.fixture(name="client")
+def client_fixture(session: Session):
+    def get_session_override():
+        return session
+
+    app.dependency_overrides[get_session] = get_session_override
```

Comment:
FastAPI's way to use an isolated (in memory) sqlite DB for testing, this goes together with the dependency injection (`session: Session = Depends(get_session)`) in the views

Reply:
Docs:
https://fastapi.tiangolo.com/tutorial/dependencies/

Also wrote about it here: 
https://pybit.es/articles/building-a-500-line-api-regression-test-suite/